compiler/rustc_trait_selection/src/traits/structural_match.rs

```diff
@@ -227,8 +217,7 @@ impl<'a, 'tcx> TypeVisitor<'tcx> for Search<'a, 'tcx> {
 
         if !self.type_marked_structural(ty) {
             debug!("Search found ty: {:?}", ty);
-            self.found = Some(NonStructuralMatchTy::Adt(&adt_def));
-            return ControlFlow::BREAK;
+            return ControlFlow::Break(NonStructuralMatchTy::Adt(&adt_def));
```

Comment:
❤️

Reply:
Yeah, I find the code so much cleaner this way 😄